src/main/java/org/spongepowered/common/registry/type/text/SelectorTypeRegistryModule.java

```diff
@@ -24,40 +24,18 @@
  */
 package org.spongepowered.common.registry.type.text;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Maps;
-import org.spongepowered.api.registry.CatalogRegistryModule;
-import org.spongepowered.api.registry.util.RegisterCatalog;
 import org.spongepowered.api.text.selector.SelectorType;
-import org.spongepowered.api.text.selector.SelectorTypes;
+import org.spongepowered.common.registry.type.MutableCatalogRegistryModule;
 import org.spongepowered.common.text.selector.SpongeSelectorType;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Optional;
-
-public final class SelectorTypeRegistryModule implements CatalogRegistryModule<SelectorType> {
-
-    @RegisterCatalog(SelectorTypes.class)
-    private final Map<String, SelectorType> selectorMappings = Maps.newHashMap();
-
-    @Override
-    public Optional<SelectorType> getById(String id) {
-        return Optional.ofNullable(this.selectorMappings.get(checkNotNull(id).toLowerCase()));
-    }
-
-    @Override
-    public Collection<SelectorType> getAll() {
-        return ImmutableList.copyOf(this.selectorMappings.values());
-    }
+public final class SelectorTypeRegistryModule extends MutableCatalogRegistryModule<SelectorType> {
```

Comment:
Improved readability, returning a `Set` instead of a `List`, reduction of boilerplate.